minifi-api/include/minifi-cpp/core/RecordField.h

```diff
@@ -30,26 +30,8 @@ namespace org::apache::nifi::minifi::core {
 
 struct RecordField;
 
-struct BoxedRecordField {
-  BoxedRecordField() = default;
-  BoxedRecordField(const BoxedRecordField&) = delete;
-  BoxedRecordField(BoxedRecordField&& rhs) noexcept : field(std::move(rhs.field)) {}
-  BoxedRecordField& operator=(const BoxedRecordField&) = delete;
-  BoxedRecordField& operator=(BoxedRecordField&& rhs)  noexcept {
-    field = std::move(rhs.field);
-    return *this;
-  };
-  ~BoxedRecordField() = default;
-
-  explicit BoxedRecordField(std::unique_ptr<RecordField>&& _field) : field(std::move(_field)) {}
-  bool operator==(const BoxedRecordField&) const;
-  std::unique_ptr<RecordField> field = nullptr;
-};
-
-
 using RecordArray = std::vector<RecordField>;
-
-using RecordObject = std::unordered_map<std::string, BoxedRecordField>;
+using RecordObject = std::unordered_map<std::string, RecordField>;
```

Comment:
Just out of curiosity, why is the boxing no longer necessary? (Or why was it previously necessary?)

Reply:
It was neccessary on centos 7, because it didnt compile without it. And now it doesnt compile with c++23 due to more restrictions on incomplete types with unique_ptr